lib/rust/parser/generate-java/java/org/enso/syntax2/Parser.java

```diff
@@ -0,0 +1,45 @@
+package org.enso.syntax2;
+
+import org.enso.syntax2.serialization.Message;
+import java.io.File;
+import java.nio.ByteBuffer;
+import java.nio.ByteOrder;
+
+public final class Parser implements AutoCloseable {
+    static {
+        File parser = new File("target/rust/debug/libenso_parser.so");
+        System.load(parser.getAbsolutePath());
+    }
+
+    private long state;
+
+    private Parser(long stateIn) {
+        state = stateIn;
+    }
+    private static native long allocState();
+    private static native void freeState(long state);
+    private static native ByteBuffer parseInput(long state, ByteBuffer input);
+    private static native long getLastInputBase(long state);
+
+    public static Parser create() {
+        var state = allocState();
+        return new Parser(state);
+    }
+    public final Tree parse(String input) {
+        try {
+            byte[] inputBytes = input.getBytes("UTF-8");
+            ByteBuffer inputBuf = ByteBuffer.allocateDirect(inputBytes.length);
+            inputBuf.put(inputBytes);
+            var serializedTree = parseInput(state, inputBuf);
```

Comment:
The call to `parseInput` shall always succeed and return _something_ even if the source code is broken. I have implemented a check loading many Enso files in our repository https://github.com/enso-org/enso/pull/3600/commits/b5a9c23ac26bc10d3d952dd0a8cc92ddcd3091a2 and it panics processing `distribution/lib/Standard/Visualization/0.0.0-dev/src/Scatter_Plot.enso`.

The ideal state is that the parsing finishes, and the error is somehow notified back to the Java method, however the Java process has to go on and try to process next file. I am interested in ration of _parseable_ and _non-parseable_ files in our repository.